src/Cake.Common.Tests/Unit/Solution/Project/XmlDoc/XmlDocExampleCodeParserTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Cake.Common.Tests.Fixtures;
+using Xunit;
+
+namespace Cake.Common.Tests.Unit.Solution.Project.XmlDoc
+{
+    public sealed class XmlDocExampleCodeParserTests
+    {
+        public sealed class TheParseMethod
+        {
+             [Fact]
+            public void Should_Throw_If_Xml_Path_Was_Null()
+            {
+                // Given
+                var fixture = new XmlDocExampleCodeParserFixture();
+                fixture.XmlFilePath = null;
+
+                // When
+                var result = Record.Exception(() => fixture.Parse());
+
+                // Then
+                Assert.IsArgumentNullException(result, "xmlFilePath");
+            }
+
+            [Fact]
+            public void Should_Throw_If_Xml_Is_Missing()
+            {
+                // Given
+                var fixture = new XmlDocExampleCodeParserFixture();
+                fixture.XmlFilePath = "/Working/missing.xml";
+
+                // When
+                var result = Record.Exception(() => fixture.Parse());
+
+                // Then
+                Assert.IsType<FileNotFoundException>(result);
+                Assert.Equal("Supplied xml file not found.", result.Message);
+            }
+
+            [Fact]
+            public void Should_Return_Two_ExampleCode()
+            {
+                // Given
+                var fixture = new XmlDocExampleCodeParserFixture();
+
+                // When
+                var result = fixture.Parse();
+
+                // Then
+                Assert.Equal(2, result.Count());
+            }
+        }
+        public sealed class TheParseFilesMethod
+        {
+             [Fact]
+            public void Should_Throw_If_Pattern_Was_Null()
+            {
+                // Given
+                var fixture = new XmlDocExampleCodeParserFixture();
+                fixture.Pattern = null;
+
+                // When
+                var result = Record.Exception(() => fixture.ParseFiles());
+
+                // Then
+                Assert.IsArgumentNullException(result, "pattern");
+            }
+
+            [Fact]
+            public void Should_Throw_If_Pattern_Is_Empty()
+            {
+                // Given
+                var fixture = new XmlDocExampleCodeParserFixture();
+                fixture.Pattern = "";
+
+                // When
+                var result = Record.Exception(() => fixture.ParseFiles());
+
+                // Then
+                Assert.IsArgumentNullException(result, "pattern");
+            }
+
+            [Fact]
+            public void Should_Return_Four_ExampleCode()
```

Comment:
Should be called something like `Should_Return_Correct_Number_Of_Examples` or similar.